src/react/Link/Link.tsx

```diff
@@ -0,0 +1,29 @@
+import type { ComponentPropsWithoutRef, JSX, ReactNode } from 'react'
+import { forwardRef } from 'react'
+
+type LinkProps<T extends React.ElementType> = {
+  children?: ReactNode
+  component?: T
+} & ComponentPropsWithoutRef<T>
+
+// export type LinkComponent = <T extends React.ElementType = 'a'>(
+//   props: LinkProps<T> & { ref?: React.ComponentPropsWithRef<T>['ref'] },
+// ) => JSX.Element
+
+export interface LinkComponent {
+  <T extends React.ElementType = 'a'>(
```

Comment:
Is there a way to block passing 'button', 'div' etc? (Didn't find one from a quick test...)